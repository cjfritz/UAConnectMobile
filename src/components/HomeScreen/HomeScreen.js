import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  Text,
  Button,
  View,
  Header,
  Right,
} from 'native-base';
import { logoutUser } from '../../actions/AuthActions';
import styles from './HomeScreen.style';
import { LoadingModal } from '../common/LoadingModal/LoadingModal';
import QuickLink from '../common/QuickLink/QuickLink';
// component to show the homescreen
export class HomeScreen extends Component {
  render() {
    // deconstruct the props from this.props
    const { userLoggedOut, loading, navigation } = this.props;
    return (
      <View style={ styles.Container }>
        <LoadingModal visible={ loading } loadingLabel='Logging out...' />
        <Header style={ styles.logout }>
          <Right>
            <Button block light onPress={ () => userLoggedOut() }>
              <Text style={ styles.logoutText }>Log Out</Text>
            </Button>
          </Right>
        </Header>
        <View style={ styles.Header }>
          <Text style={ styles.uaconnect }> UA Connect </Text>
        </View>
        <View style={ styles.BContainer }>
          <View style={ styles.Row1 }>
            <QuickLink
              iconName='ios-person'
              color='rgba(255, 255, 255, 0.90)'
              onPress={ () => navigation.navigate('Profile') }
            >
              Profile
            </QuickLink>
            <QuickLink
              iconName='ios-bookmarks'
              color='rgba(255, 255, 255, 0.90)'
              onPress={ () => navigation.navigate('Planner') }
            >
              Course Planner
            </QuickLink>
          </View>
          <View style={ styles.Row2 }>
            <QuickLink
              iconName='ios-calendar'
              color='rgba(255, 255, 255, 0.90)'
              onPress={ () => navigation.navigate('Events') }
            >
              Events
            </QuickLink>
            <QuickLink
              iconName='md-paper'
              color='rgba(255, 255, 255, 0.90)'
              onPress={ () => navigation.navigate('News') }
            >
              News
            </QuickLink>
          </View>
        </View>
      </View>
    );
  }
}
// map the action creators to component props
const mapDispatchToProps = dispatch => (
  ({
    userLoggedOut: () => dispatch(logoutUser()),
  })
);
// map the app state to component props
const mapStateToProps = state => {
  const { loading } = state.auth;

  return { loading };
};
// export the homescreen component
export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

