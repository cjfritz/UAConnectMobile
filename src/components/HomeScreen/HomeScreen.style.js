import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  BContainer: {
    flexGrow: 7,
  },
  Container: {
    alignContent: 'center',
    backgroundColor: '#67141A',
    flex: 1,
  },
  Head: {
    backgroundColor: '#ffffff',
  },
  Header: {
    alignItems: 'center',
    backgroundColor: '#67141A',
    flex: 2,
    justifyContent: 'center',
  },
  Row1: {
    backgroundColor: '#67141A',
    flex: 1,
    flexDirection: 'row',
  },
  Row2: {
    backgroundColor: '#67141A',
    flex: 1,
    flexDirection: 'row',
  },
  logout: {
    backgroundColor: 'darkred',
    color: 'black',
    fontSize: 60,
  },
  logoutText: {
    color: 'black',
    fontSize: 15,
  },
  uaconnect: {
    color: 'white',
    fontSize: 60,
  },
});
