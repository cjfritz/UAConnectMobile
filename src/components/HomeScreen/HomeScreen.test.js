import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { HomeScreen } from './HomeScreen';

const props = {
  loading: false,
  userLoggedOut: jest.fn(),
};

describe('<Homescreen />', () => {
  it('renders with default props', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<HomeScreen { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });
});
