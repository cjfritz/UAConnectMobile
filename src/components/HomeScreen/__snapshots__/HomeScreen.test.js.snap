// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homescreen /> renders with default props 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#67141A",
      "flex": 1,
    }
  }
>
  <LoadingModal
    loadingLabel="Logging out..."
    visible={false}
  />
  <Styled(Header)
    style={
      Object {
        "backgroundColor": "darkred",
        "color": "black",
        "fontSize": 60,
      }
    }
  >
    <Styled(Right)>
      <Styled(Button)
        onPress={[Function]}
        transparent={true}
      >
        <Styled(Text)
          style={
            Object {
              "color": "white",
              "fontSize": 15,
            }
          }
        >
           Log Out 
        </Styled(Text)>
      </Styled(Button)>
    </Styled(Right)>
  </Styled(Header)>
  <Styled(ViewNB)
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#67141A",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "white",
          "fontSize": 60,
        }
      }
    >
       UA Connect 
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "flexGrow": 7,
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#67141A",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <QuickLink
        color="rgba(255, 255, 255, 0.90)"
        iconName="ios-person"
        onPress={[Function]}
      >
        Profile
      </QuickLink>
      <QuickLink
        color="rgba(255, 255, 255, 0.90)"
        iconName="ios-bookmarks"
        onPress={[Function]}
      >
        Course Planner
      </QuickLink>
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#67141A",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <QuickLink
        color="rgba(255, 255, 255, 0.90)"
        iconName="ios-calendar"
        onPress={[Function]}
      >
        Events
      </QuickLink>
      <QuickLink
        color="rgba(255, 255, 255, 0.90)"
        iconName="md-paper"
        onPress={[Function]}
      >
        News
      </QuickLink>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
